### DISCUSSION TITLE
.DDS texture support on Android?

### DISCUSSION THREAD

**Discussion by george7378** [MonoGame/MonoGame#8475] [Android]
Hi everyone - I'm wondering if someone knows about DDS texture format support on Android devices?

I'm loading a DDS cubemap texture into my game with **no compression** - which is loaded into my MGCB file like this:

`#begin ../../path/to/cubeMap.dds
/importer:TextureImporter
/processor:TextureProcessor
/processorParam:ColorKeyColor=255,0,255,255
/processorParam:ColorKeyEnabled=True
/processorParam:GenerateMipmaps=True
/processorParam:PremultiplyAlpha=True
/processorParam:ResizeToPowerOfTwo=False
/processorParam:MakeSquare=False
/processorParam:TextureFormat=Color
/build:../../path/to/cubeMap.dds;Textures/cubeMap.dds`

It works well on all the devices and emulators I've tried. I did initially export it with DXTn compression but this always generated 'texture format not supported' exceptions so I went with uncompressed instead, which seems fine.

So just wondering, should an uncompressed DDS cubemap be OK on all android devices (I'm targeting API 21+), or are there other compatibility things I should be aware of for this format?

**Comment by ThomasFOG** [comment]
DDS is the input format. Once it is processed by the content pipeline, it is no more DDS and the texture format is going to be what you specified in ```processorParam:TextureFormat```. Which in your case is ```Color``` (i.e. uncompressed, therefore maximum compatibility). So even if your DDS contained compressed data and it were accepted by the tool, the resulting would be uncompressed anyway.

For Android, you can select ETC for your ```TextureFormat```, which is basically supported throughout all devices safely.

**Reply by george7378** [reply]
Ah right, got it, thanks!

**Reply by dellis1972** [reply]
@george7378 just a heads up ETC1 does not support RGBA, ETC2 does, but it might not be supported on all devices. 
I would recommend you set the texture to be `Compressed` and it will default to the RGBA4444 if it needs an Alpha channel or RGB565 if its just RGB. 

If you really want the best compression you will need to provide textures in ETC2, ATITC, PVRTC and RGBA4444 at a minimum. This is because there are so many android devices with different GPU's from different manufacturers.
